convert2rhel/subscription.py

```diff
@@ -242,13 +246,33 @@ def install_rhel_subscription_manager():
         # When using the original system repos, we need YUM/DNF to expand the $releasever by itself
         set_releasever=False,
     )
-    if ret_code:
-        loggerinst.critical("Failed to install subscription-manager packages. See the above yum output for details.")
-    else:
-        loggerinst.info("Packages installed:\n%s" % "\n".join(rpms_to_install))
-        pkg_names = get_installed_submgr_pkg_names(rpms_to_install)
-        utils.changed_pkgs_control.track_installed_pkgs(pkg_names)
+    if not ret_code:
+        loggerinst.info("\nPackages installed:\n%s" % "\n".join(rpms_to_install))
+
+        # Remove any packages that are present in the system before the install
+        [pkg_names.remove(rpm) for rpm in rpms_to_untrack if rpm in pkg_names]
+        loggerinst.debug("Untracked pacakges: %r" % rpms_to_untrack)
         loggerinst.debug("Tracking installed packages: %r" % pkg_names)
+        utils.changed_pkgs_control.track_installed_pkgs(pkg_names)
+    else:
+        loggerinst.critical("Failed to install subscription-manager packages. See the above yum output for details.")
+
+
+def check_for_pkg_presence(pkg_names):
+    """Returns a list of packages that are already present in the system before
+    installing any packages needed by subscrpition-manager
```

Comment:
```suggestion
    """Check that packages are installed on the system.
```

Reply:
Reason for this change is because the function is unrelated to subscription-manager and is very generic